### PR TITLE
Add RT-DETRv2-S object detection sample (CompiledModel GPU, 2-graph split)

### PR DESCRIPTION
**Pull request by john-rocky**
Adds **RT-DETRv2-S** ([lyuwenyu/RT-DETR](https://github.com/lyuwenyu/RT-DETR), `PekingU/rtdetr_v2_r18vd`, Apache-2.0) to the `compiled_model_api/object_detection` example, alongside YOLOX and RF-DETR — a **transformer/DETR** detector running 100% on the LiteRT `CompiledModel` GPU. **Still-image** demo (see timing below).

> **Stacked on #166 (YOLOX) and #180 (RF-DETR).** Please merge those first; the only new commits here are the RT-DETRv2 sample (`compiled_model_api/object_detection/rt_detr_v2_kotlin_gpu/`).

## Why it's interesting
RT-DETRv2 is a two-stage DETR; the query selection is `TOPK`/`GATHER`, which have no GPU op. It's converted with **litert-torch** and split into **two GPU graphs** around the query selection, with the per-token tail on the host:

```
image →[GPU Graph A]→ enc_class, memory_raw
      →[host: top-300; per-token tail on the 300: target = enc_output(valid·memory_raw); ref = bbox_head(target)+anchors]
      →[GPU Graph B (memory_raw, target, ref)]→ boxes, logits
      →[host: sigmoid + threshold + cxcywh→xyxy + light NMS]→ detections
```

## The on-device gate — a Mali 3D-sequence fan-out bug (not an fp16 wall)
A naïve Graph A (emitting `enc_class`/`enc_coord`/`output_memory`/`memory_raw` together) silently produced wrong boxes on device — large objects vanished, small ones stayed perfect. A **3D token tensor `[1,N,256]`** (from `conv.flatten(2).transpose(1,2)`) that is output-and-consumed / fans out inside one GPU graph gets clobbered on the longer branch (4D conv-map outputs are fine). `output_memory` fed both heads; the 3-layer box head lost, so its reference deltas collapsed to ~0 (masked in correlation by the ±1e4 anchors). **Fix:** Graph A emits only the two fp16-clean leaves (`enc_class` + `memory_raw*2`); the per-token tail runs on the host in fp32 over the 300 selected tokens (exact, since per-token ops commute with gather). Other re-authoring (corr 1.0): deformable `grid_sample` → tent-matmul, MSDeformAttn ≤4D, SafeLayerNorm, ResNet18-vd stem zero-pad maxpool, baked AIFI sine pos-embed.

## On-device (Pixel 8a, Tensor G3 — verified)
- Both graphs fully on `LITERT_CL` (Graph B `704/704`); the device chain reproduces the PyTorch detections **exactly** — COCO val giraffe 7/7, cats 6/6, **IoU 0.98–1.00**.
- End-to-end **~615 ms/frame**: RT-DETR's 8400-token / 80×80 deformable decoder is ~350 ms of GPU compute (the GATHER-free tent-matmul is the cost), so this is accurate but not real-time — hence a still-image demo. (RF-DETR, one small 24×24 deformable level, is the real-time camera sample, #180.)

Models: [litert-community/RT-DETRv2-S-LiteRT](https://huggingface.co/litert-community/RT-DETRv2-S-LiteRT).

### CODE REVIEW

**Comment by john-rocky** [comment]
Closing to consolidate the object detection samples (as proposed in our collaboration doc): keeping one DETR variant only — RF-DETR (#180), which runs real-time on device (~110ms on Pixel 8a), while RT-DETRv2-S is still-image class (~615ms). The model itself remains available at litert-community/RT-DETRv2-S-LiteRT with the conversion script.
